spec/compiler/formatter/formatter_spec.cr

```diff
@@ -1140,4 +1140,11 @@ describe Crystal::Formatter do
   assert_format "foo do\n  bar do\n    foo <<-X\n    bar\n    X\n  end\nend"
   assert_format "call(foo <<-X\nbar\nX\n)"
   assert_format "bar do\n  call(foo <<-X\n  bar\n  X\n  )\nend"
+
+  assert_format "[\n  <<-EOF,\n  foo\n  EOF\n]"
+  assert_format "[\n  <<-EOF,\n  foo\n  EOF\n  <<-BAR,\n  bar\n  BAR\n]"
+  assert_format "Hash{\n  foo => <<-EOF,\n  foo\n  EOF\n}"
+  assert_format "Hash{\n  foo => <<-EOF,\n  foo\n  EOF\n  bar => <<-BAR,\n  bar\n  BAR\n}"
+  assert_format "Hash{\n  foo => <<-EOF\n  foo\n  EOF\n}"
```

Comment:
Is this really right? I think it should append comma. So correct assertion is:

```crystal
assert_format "Hash{\n  foo => <<-EOF\n  foo\n  EOF\n}", "Hash{\n  foo => <<-EOF,\n  foo\n  EOF\n}"
```

Because this code:

```crystal
{
  foo: 42
}
```

is formatted to:

```crystal
{
  foo: 42,
}
```

Reply:
Yeah, that's probably better. *But*  it seems a bit more difficult because of the special treatment of newline in `next_token`. And it requires changes in a some other examples, that are - currently - explicitly not adding a comma there:

https://github.com/straight-shoota/crystal/blob/4c63cd398fdf2f98ee05a6e12506e979d692c5ab/spec/compiler/formatter/formatter_spec.cr#L1135-L1138

I'd suggest to leave this PR as a fix for the immediate issue.

Reply:
I agree with @straight-shoota here. Not adding a comma although is inconsistent with non heredoc hash/array literals, at least is not producing invalid code.

@sdogruyol @MakeNowJust WDYT?

Reply:
Okay.

Reply:
I think, this is OK :+1: